lib/api/include/irods/check_auth_credentials.h

```diff
@@ -0,0 +1,131 @@
+#ifndef IRODS_CHECK_AUTH_CREDENTIALS_H
+#define IRODS_CHECK_AUTH_CREDENTIALS_H
+
+/// \file
+
+struct RcComm;
+
+/// The input data type used to invoke #rc_check_auth_credentials.
+///
+/// \since 4.3.1
+typedef struct CheckAuthCredentialsInput // NOLINT(modernize-use-using)
+{
+    // NOLINTBEGIN(cppcoreguidelines-avoid-c-arrays, modernize-avoid-c-arrays)
+    // NOLINTBEGIN(cppcoreguidelines-avoid-magic-numbers, readability-magic-numbers)
+
+    /// The part of a fully-qualified iRODS username preceding the pound sign.
+    ///
+    /// The following requirements must be satisfied:
+    /// - The length must not exceed sizeof(UserInfo::userName) - 1 bytes
+    /// - It must be null-terminated
+    /// - It must be non-empty
+    /// - It must not include the pound sign
+    /// - It must not include the zone
+    ///
+    /// \since 4.3.1
+    char username[64];
+
+    /// The part of a fully-qualified iRODS username following the pound sign.
+    ///
+    /// The following requirements must be satisfied:
+    /// - The length must not exceed sizeof(UserInfo::rodsZone) - 1 bytes
+    /// - It must be null-terminated
+    /// - It must be non-empty
+    /// - It must not include the pound sign
+    /// - It must not include the username
+    ///
+    /// \since 4.3.1
+    char zone[64];
+
+    /// The expected obfuscated password for the user of interest.
+    ///
+    /// The following requirements must be satisfied:
+    /// - The length must not exceed 250 - 1 bytes
+    /// - It must be null-terminated
+    /// - It must be non-empty
+    ///
+    /// \since 4.3.1
+    char password[250];
+
+    // NOLINTEND(cppcoreguidelines-avoid-magic-numbers, readability-magic-numbers)
+    // NOLINTNEND(cppcoreguidelines-avoid-c-arrays, modernize-avoid-c-arrays)
+} checkAuthCredsInp_t;
+
+// NOLINTNEXTLINE(cppcoreguidelines-macro-usage)
+#define CheckAuthCredentialsInput_PI "str username[64]; str zone[64]; str password[250];"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/// Verifies the native authentication credentials passed match the credentials stored
+/// in the catalog.
+///
+/// This API endpoint only supports native authentication and requires rodsadmin
+/// privileges.
+///
+/// It does NOT modify any state on either side of the connection.
+///
+/// \param[in] _comm    A pointer to a RcComm.
+/// \param[in] _input   A pointer to a CheckAuthCredentialsInput.
+/// \param[in] _correct \parblock A pointer to an integer pointer used to indicate
+/// whether the native authentication credentials are correct.
+///
+/// On success, the pointer will point to memory on the heap. The caller is
+/// responsible for freeing the memory.
```

Comment:
What is the reason we aren't just using an `int*` here? I don't remember if this was discussed...

Reply:
Returning values from the server requires a pointer-to-pointer parameter.

I normally hide that detail, but I chose not to for this. Usage is identical to that of `rxDataObjChksum`.

Reply:
Before these latest commits, I updated things to use an `int*`, but that didn't sit well with me because of PackStruct. While I did not run that code, I didn't have confidence things would work that way just based on what I knew about PackStruct.

Still, pointer-to-pointer parameters aren't anything unfamiliar to C devs.

Reply:
Ah, right. Sounds good.